onix-bot/onixparser.py

```diff
@@ -0,0 +1,223 @@
+
+    def test_publication_country(self):
+        publication_country = self.op.products[0].publication_country
+        expected_publication_country = "GB"
+        
+        self.assertTrue(expected_publication_country == publication_country)
+
+    def test_publication_city(self):
+        publication_city = self.op.products[0].publication_city
+        expected_publication_city = "Oxford"
+        
+        self.assertTrue(expected_publication_city == publication_city)
+
+    def test_json(self):
+        onix_json = self.op.products[0].get_json
+        expected_onix_json = json.dumps({"title": "Roman Art", "publication_country": "GB", "publication_city": "Oxford", "identifiers": {"isbn10": "0199223955", "isbn13": "9780199223954"}, "authors": "", "publishers": "Oxford University Press", "languages": "eng"})
+
+        self.assertTrue(expected_onix_json == onix_json)
+
+
+class OnixFeedParser(object):
+
+    # def __init__(self, filename, ns="http://www.editeur.org/onix/2.1/reference"):
+    def __init__(self, filename, ns=""):
+        parser = etree.XMLParser(ns_clean=True)
+        self.onix = etree.parse(filename, parser).getroot()
+        self.ns = ns
+        # self.products = [OnixProductParser(product, ns) for product in self.onix.findall('{%s}Product' % ns)]
+        self.products = [OnixProductParser(product, ns) for product in self.onix.findall('Product')]
+        # self.errors = onixcheck.validate(filename.getvalue())
+
+
+class OnixProductParser(object):
+
+    def __init__(self, product, ns):
+        self.ns = ns
+        self.product = product
+
+    @property
+    def title(self):
+        # title = self.product.xpath('//ns:Title', namespaces={'ns': self.ns})
+        title = self.product.xpath('//Title')
+        if title:
+            # title = title[0].xpath('//ns:TitleText', namespaces={'ns': self.ns})
+            title = title[0].xpath('//TitleText')
+        return title[0].text if title else ''
+
+    @property
+    def publisher(self):
+        # publisher = self.product.xpath('//ns:Publisher', namespaces={'ns': self.ns})
+        publisher = self.product.xpath('//Publisher')
+        if publisher:
+            # publisher = publisher[0].xpath('//ns:PublisherName', namespaces={'ns': self.ns})
+            publisher = publisher[0].xpath('//PublisherName')
+        return publisher[0].text if publisher else ''
+
+    @property
+    def authors(self):
+        # authors = self.product.xpath('//ns:Author', namespaces={'ns': self.ns})
+        authors = self.product.xpath('//Author')
+        
+        book_authors = []
+
+        if authors:
+            for author in authors:
+                book_authors.append(author[1].text)
+
+        return book_authors if authors else ''
+
+    @property
+    def languages(self):
+        # languages = self.product.xpath('//ns:Language', namespaces={'ns': self.ns})
+        languages = self.product.xpath('//Language')
+        
+        if languages:
```

Comment:
here, the function is called `languages` and so we'd expect that it returns either an empty list or a list of languages:

`return [language[0].text for language in languages]`